### PR TITLE
Pbf 550

### PR DESCRIPTION
**Pull request by sarita-ooyala**

### CODE REVIEW

File: src/akamai/lib/AkamaiHDPlayer.as [hunk 1]
```diff
@@ -406,7 +424,26 @@ package
...
+        for (var i:int = 0; i < getStreamsCount(); i++)
+        {
+          _bitrateIdArray.push((_streamController.mediaPlayer.getBitrateForDynamicStreamIndex(i)) + "kbps");
+          id = _bitrateIdArray[i];
+          var bitrateObject:Object = new Object();
+          bitrateObject.id = id;
+          bitrateObject.height = 0;
+          bitrateObject.width = 0;
+          bitrateObject.bitrate = _streamController.mediaPlayer.getBitrateForDynamicStreamIndex(i) * 1000;
+          _bitrateArray[id] = [ bitrateObject, i];
```
Comment (chibzu) [inline src/akamai/lib/AkamaiHDPlayer.as]: This is being used as a map and not an Array, isn't it? The "id" variable is a String so you're not indexing into it, you're setting key value pairs on it as if it were an Object. Array may not be the clearest data type here for _bitrateArray.

File: src/akamai/js/akamaiHD_flash.js [hunk 2]
```diff
@@ -485,50 +487,62 @@ require("../../../html5-common/js/utils/constants.js");
...
+      self.controller.notify(self.controller.EVENTS.BITRATES_AVAILABLE,vtcBitrates);
     };
 
     var raiseSizeChanged = function(event) {
     };
 
     // Receives a callback from Flash 
     onCallback = _.bind(function(data) {
-      OO.log("[AKAMIHD]:onCallback: ", data);
+     OO.log("[AKAMIHD]:onCallback: ", data);
```
Comment (chibzu) [inline src/akamai/js/akamaiHD_flash.js]: inconsistent indentation

File: src/akamai/js/akamaiHD_flash.js [hunk 3]
```diff
@@ -485,50 +487,62 @@ require("../../../html5-common/js/utils/constants.js");
...
       switch (eventtitle)
       {
-       case "JSREADY":
-        for (i = 0; i < actionscriptCommandQueue.length; i++) {
-          this.callToFlash(actionscriptCommandQueue[i][0],actionscriptCommandQueue[i][1]);
+        case "JSREADY":
+         while(0 < actionscriptCommandQueue.length) {
+          this.callToFlash(actionscriptCommandQueue[0][0],actionscriptCommandQueue[0][1]);
+          actionscriptCommandQueue.shift();
         }
```
Comment (chibzu) [inline src/akamai/js/akamaiHD_flash.js]: inconsistend indentation

**Comment by angieisrani** [comment]
@sarita-ooyala This PR contains changes from PBF-543 - if possible, please separate these or wait until PBF-543 is merged before opening this

File: src/akamai/lib/AkamaiHDPlayer.as [hunk 4]
```diff
@@ -167,6 +169,7 @@ package
         case MediaPlayerState.LOADING:
           break;
         case MediaPlayerState.READY:
+          totalBitratesAvailable();
```
Comment (angieisrani) [inline src/akamai/lib/AkamaiHDPlayer.as]: Would you mind renaming this function to show what it does?  It provides the bitrate information to the controller

File: src/akamai/lib/AkamaiHDPlayer.as [hunk 5]
```diff
@@ -406,7 +424,26 @@ package
      * @method AkamaiHDPlayer#totalBitratesAvailable
      */
     public function  totalBitratesAvailable():void
-    { 
+    {
+      if (_bitrateIdArray.length > 0 ) return;
+      var eventObject:Object = new Object();
+      var id:String;
+      if (getStreamsCount() > 0)
```
Comment (angieisrani) [inline src/akamai/lib/AkamaiHDPlayer.as]: small note that this if could just be around the dispatch but this works too

**Comment by sarita-ooyala** [comment]
Changes are done as per comments. Please merge after PBF-543 is  get merged .

File: src/akamai/lib/AkamaiHDPlayer.as [hunk 6]
```diff
@@ -92,6 +94,32 @@ package
...
+    private function SendToDebugger(value:String, referrer:String = null, channelBranch:String = "log"):Boolean
+    {
+      var channel:String; 
+      if (channelBranch == "log") 
+      {
+        channel = "OO." + channelBranch;
+      }
+      else 
+      {
+        channel = "console.log";
```
Comment (angieisrani) [inline src/akamai/lib/AkamaiHDPlayer.as]: So we shouldn't actually be writing to console.log ever.  You can lot go console.error and console.warn, but if the message is type log it should be OO.log.  Looks like you are mapping all non-log messages back to console.log?  Maybe try "console." + channelBranch

File: src/akamai/lib/ExternalJavaScriptAPI.as [hunk 7]
```diff
@@ -85,7 +85,6 @@ package
       addEventListener("setVideoClosedCaptions", onSetVideoClosedCaptions);
       addEventListener("setVideoClosedCaptionsMode", onSetVideoClosedCaptionsMode);
       addEventListener("setTargetBitrate", onSetTargetBitrate);
-      addEventListener("setInitialTime", onSetInitialTime);
```
Comment (angieisrani) [inline src/akamai/lib/ExternalJavaScriptAPI.as]: was this removed on purpose?

Comment (sarita-ooyala) [inline src/akamai/lib/ExternalJavaScriptAPI.as]: Yes, we are calling setInitialTime in seek event. So no need of separate setInitialTime event

**Comment by chibzu** [comment]
Looks good, just needs conflicts resolved.

**Comment by angieisrani** [comment]
@chibzu  if you merge, do not merge this before PBF-543, this PR contains commits from that branch as well

**Comment by angieisrani** [comment]
@sarita-ooyala Please resolve the conflicts here

**Comment by sarita-ooyala** [comment]
Resolved merge conflicts

**Pull request merged**